File: 0011-Container With Most Water/0011-Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxA = 0;
        int start = 0, end = height.size() - 1;
        while (start < end)
        {
            maxA = max(maxA, min(height[start], height[end]) * (end -start));
            if (height[start] <= height[end])
            {
                ++start;
            }
            else
            {
                --end;
            }
        }
        
        return maxA;
    }
};

File: 0001-0100/0011-Container With Most Water/0011-Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        maxA = 0
        while left < right:
            maxA = max(maxA, min(height[left], height[right])*(right - left))
            if height[left] <= height[right]:
                left += 1
            else:
                right -= 1
        return maxA
